SDL/TrackCandidate.cu

```diff
@@ -339,7 +347,7 @@ __global__ void SDL::addpT3asTrackCandidatesInGPU(struct SDL::pixelTriplets& pix
         atomicAdd(trackCandidatesInGPU.nTrackCandidatespT3,1);
     
         float radius = 0.5f * (__H2F(pixelTripletsInGPU.pixelRadius[pixelTripletIndex]) + __H2F(pixelTripletsInGPU.tripletRadius[pixelTripletIndex]));
-        addTrackCandidateToMemory(trackCandidatesInGPU, 5/*track candidate type pT3=5*/, pixelTripletIndex, pixelTripletIndex, &pixelTripletsInGPU.logicalLayers[5 * pixelTripletIndex], &pixelTripletsInGPU.lowerModuleIndices[5 * pixelTripletIndex], &pixelTripletsInGPU.hitIndices[10 * pixelTripletIndex], __H2F(pixelTripletsInGPU.centerX[pixelTripletIndex]), __H2F(pixelTripletsInGPU.centerY[pixelTripletIndex]),radius,trackCandidateIdx);
+        addTrackCandidateToMemory(trackCandidatesInGPU, 5/*track candidate type pT3=5*/, pixelTripletIndex, pixelTripletIndex, &pixelTripletsInGPU.logicalLayers[5 * pixelTripletIndex], &pixelTripletsInGPU.lowerModuleIndices[5 * pixelTripletIndex], &pixelTripletsInGPU.hitIndices[10 * pixelTripletIndex], __H2F(pixelTripletsInGPU.centerX[pixelTripletIndex]), __H2F(pixelTripletsInGPU.centerY[pixelTripletIndex]),radius,trackCandidateIdx, pixelTripletIndex);
```

Comment:
I can see that, in most cases, the additional argument is also the second argument in the same function call, so we could perhaps save ourselves from adding a new argument?

The above doesn't seem to be the case only for the pT5s but this looks like an inconsistency to me that may need addressing.

Reply:
I must have not noticed that that variable was already passed on...
It doesn't help the function call is already reallly long...
Let me see if i can rever some of these.

Reply:
addTrackCandidateToMemory(trackCandidatesInGPU, 7/*track candidate type pT5=7*/, pixelQuintupletsInGPU.pixelIndices[pixelQuintupletIndex], pixelQuintupletsInGPU.T5Indices[pixelQuintupletIndex], &pixelQuintupletsInGPU.logicalLayers[7 * pixelQuintupletIndex], &pixelQuintupletsInGPU.lowerModuleIndices[7 * pixelQuintupletIndex], &pixelQuintupletsInGPU.hitIndices[14 * pixelQuintupletIndex], __H2F(pixelQuintupletsInGPU.centerX[pixelQuintupletIndex]),
                            __H2F(pixelQuintupletsInGPU.centerY[pixelQuintupletIndex]),radius , trackCandidateIdx, pixelQuintupletIndex);

Reply:
Ah I remember now.
So L194 and L350 needs to be called through a same function, and in one case, it is a duplicate information, but in the other it is not.

Fundamentally the problem is due to the fact that there is some sort of harmonization needed between how we treat pT3 and pT5.

Reply:
Is there an action item on this? Something concrete that we could make an issue of and fix it in a new PR?

In any case, I don't think the questions is related to this PR anymore.